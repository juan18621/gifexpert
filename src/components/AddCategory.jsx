import { useState } from "react";
import PropTypes from 'prop-types'


export const AddCategory = ({onAddCategory}) => {


    const  [inputValue, setInputValue] = useState('One Punch')

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit  = (event) => {
        event.preventDefault()
        if(inputValue.trim().length < 1){
            return
        }
        onAddCategory(inputValue.trim())
        setInputValue('');
    }

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input type="text" placeholder="Buscar gifs" value={inputValue} onChange={onInputChange}/>
        </form>
    );
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired
}