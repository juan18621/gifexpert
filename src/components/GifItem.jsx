import PropTypes from 'prop-types'


export const GifItem = ({image}) => {
  return (
    <div className="card">
        <img src={image.url} alt={image.title} />
        <p>{image.title}</p>
    </div>
  )
}

GifItem.propTypes = {
  image: PropTypes.object.isRequired
}
