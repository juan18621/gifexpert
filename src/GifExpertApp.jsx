import { useState } from "react";
import { GifGrid, AddCategory } from "./components";

export const GiftExpertApp = () => {

    const [categories, setCategories] = useState(['One Puch', 'Dragon Ball'])

    const addCategory = (newCategory) => {
        categories.includes(newCategory) ? null: setCategories([newCategory, ...categories])
    }
    return (
        <>
        <h1>GifExpertApp</h1>
        <AddCategory onAddCategory={addCategory} />
        <ol>
            {categories.map(category => <GifGrid key={new Date().getTime() + Math.random()} category={category}/> )}
        </ol>
        </>
    );
}