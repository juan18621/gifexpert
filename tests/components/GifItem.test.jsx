import {render, screen} from '@testing-library/react'
import { GifItem } from '../../src/components'


describe('pruebas en gif item', () => { 
    const url = 'https://holamundo.com/image.jpg'
    const title = ''
    test('debe hacer match con el sapshot', () => { 

        const  {container} = render(<GifItem image={{title, url}}></GifItem>)
        expect(container).toMatchSnapshot();
     })

     test('url y alt indicado', () => { 
        render(<GifItem image={{title, url}}></GifItem>);
        //screen.debug();
        expect( screen.getByRole('img').src).toBe(url)
        expect( screen.getByRole('img').alt).toBe(title)

    })

    test('debe mostrar el titulo', () => { 
        render(<GifItem image={{title, url}}></GifItem>);
        screen.debug();
        expect( screen.getAllByText(title)).toBeTruthy();
     })
 })

