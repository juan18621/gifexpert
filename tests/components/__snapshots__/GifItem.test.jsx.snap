// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en gif item debe hacer match con el sapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt=""
      src="https://holamundo.com/image.jpg"
    />
    <p />
  </div>
</div>
`;
